### PR TITLE
feat(api): support access token rotate API

### PR DESCRIPTION
**Pull request by nejch**
## Changes

Adds a `RotateMixin` and `ObjectRotateMixin` similar to the delete mixins.

Closes https://github.com/python-gitlab/python-gitlab/issues/2589
Closes https://github.com/python-gitlab/python-gitlab/issues/2746

### Documentation and testing

Please consider whether this PR needs documentation and tests. **This is not required**, but highly appreciated:

- [x] Documentation in the matching [docs section](https://github.com/python-gitlab/python-gitlab/tree/main/docs)
- [x] [Unit tests](https://github.com/python-gitlab/python-gitlab/tree/main/tests/unit) and/or [functional tests](https://github.com/python-gitlab/python-gitlab/tree/main/tests/functional)

### CODE REVIEW

File: docs/gl_objects/personal_access_tokens.rst [hunk 1]
```diff
@@ -52,6 +52,13 @@ Revoke the personal access token currently used::
 
     gl.personal_access_tokens.delete("self")
 
+Rotate a personal access token and retrieve its new value::
+
+    gl.personal_access_tokens.rotate(42)
```
Comment (max-wittig) [inline docs/gl_objects/personal_access_tokens.rst:57]: isn't that a bit confusing.

In my mind it should be `gl.personal_access_tokens.get(42, lazy=True).rotate()`

Comment (nejch) [reply docs/gl_objects/personal_access_tokens.rst]: Done by switching the docs to show the object method first.

File: gitlab/mixins.py [hunk 2]
```diff
@@ -653,6 +653,62 @@ def download(
...
+
+        Args:
+            id: ID of the token to rotate
+            **kwargs: Extra options to send to the server (e.g. sudo)
+
+        Raises:
+            GitlabAuthenticationError: If authentication is not correct
+            GitlabRotateError: If the server cannot perform the request
+        """
+        data = {"expires_at": expires_at}
```
Comment (JohnVillalovos) [inline gitlab/mixins.py:679]: Is it okay for this to be set to `None`?

Comment (nejch) [reply gitlab/mixins.py]: I tested this before making the PR and it works, but maybe we shouldn't send null parameters if the user doesn't send them just to keep requests cleaner. I changed this in https://github.com/python-gitlab/python-gitlab/commit/4f27e365de83e92804dbbfcc075f4b3c8a7a15dd, thanks!

**Pull request merged**